File: Week_01/G20190343020279/LeetCode_189_0279 .py
'''
方法：“三次反转”
'''
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        #获取数组长度
        n = len(nums)
        #忽略重复后实际移动次数
        k %= n
        #反转方法
        def swap(l,r):
            while(l<r):
                nums[l],nums[r] = nums[r],nums[l]
                l = l + 1
                r = r - 1
        #整体反转
        swap(0,n - 1)
        #前K个反过来
        swap(0,k - 1)
        #后n-k个反过来，完成
        swap(k,n - 1)

File: Week_01/G20190343020279/LeetCode_189_0279 .js
/**
 * @方法：“暴力枚举”
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    //临时变量（用来交互数字位置），前面的变量
    let temp,previous
    for (let i = 0; i < k; i++) {
        previous = nums[nums.length - 1]
        for (let j = 0; j <nums.length; j++) {
            temp = nums[j]
            nums[j] = previous
            previous = temp
        }
    }
    return nums;
};